### PR TITLE
Added resources snippets for JSON and JSONC files

### LINKED ISSUE #477: Add resources snippets for JSON and JSONC files
Snippets for resources currently exist for PowerShell and YAML files, however no snippets exist for JSON and JSON with comments files.

These should be added.

### PR DESCRIPTION
**Pull request by ArmaanMcleod**
## PR Summary

Fixes #477.

Added JSON snippets. 

Made some changes to the snippet names and descriptions between YAML and JSON to differentiate easier. 

## PR Checklist

- [x] PR has a meaningful title
- [x] Summarized changes
- [x] Change is not breaking
- [x] This PR is ready to merge and is not **Work in Progress**
- **Code changes**
  - [x] Link to a filed issue
  - [x] [Change log](https://github.com/Microsoft/PSRule-vscode/blob/main/CHANGELOG.md) has been updated with change under unreleased section

### CODE REVIEW

File: snippets/yaml.json [hunk 1]
```diff
@@ -70,7 +70,7 @@
             "metadata:",
             "  name: ${1}",
             "spec:",
-            "  selector:",
+            "  with:",
```
Comment (ArmaanMcleod) [inline snippets/yaml.json:73]: Changed this from `selector:` to `with:`.

**Review by BernieWhite — CHANGES_REQUESTED** [review]
@ArmaanMcleod Looks good. Just a few suggestions.

**Comment by ArmaanMcleod** [comment]
> @ArmaanMcleod Looks good. Just a few suggestions.

Thanks @BernieWhite. I've applied your suggestions. Also added quotes to `with:` and `type:` for the preconditions.

**Review by BernieWhite — APPROVED** [review]
@ArmaanMcleod Nice work! All good to merge.

**Pull request merged**